File: Java/src/main/java/com/adaptionsoft/games/uglytrivia/Places.java
package com.adaptionsoft.games.uglytrivia;

public class Places extends CurrentArray<Integer> {

    PlayerNames names;

    public Places(CurrentPlayer currentPlayer, PlayerNames names) {
        super(currentPlayer, 0);
        this.names = names;
    }

    public void add(int roll) {
        set(currentPlayer.get(), current() + roll);
    }

    public void printlnPlace() {
        System.out.println(names.get() 
                + "'s new location is " 
                + current());
    }
    
}

File: Csharp/Trivia/Places.cs
using System;

namespace UglyTrivia
{
    public class Places : CurrentArray<int>
    {
        PlayerNames names;

        public Places(CurrentPlayer currentPlayer, PlayerNames names)
            : base(currentPlayer, 0)
        {
            this.names = names;
        }

        public void Add(int roll)
        {
            Set(currentPlayer.Get(), Current() + roll);
        }

        public void PrintlnPlace()
        {
            Console.WriteLine(names.Get()
                + "'s new location is "
                + Current());
        }
    }
}
